smac_planner/src/smac_planner.cpp

```diff
@@ -239,22 +267,33 @@ nav_msgs::msg::Path SmacPlanner::createPlan(
   steady_clock::time_point a = steady_clock::now();
 #endif
 
+  // Choose which costmap to use for the planning
+  nav2_costmap_2d::Costmap2D * costmap;
+  if (_costmap_downsampler) {
+    costmap = _costmap_downsampler->downsample();
+    if (_node->count_subscribers(_costmap_topic_name) > 0) {
+      _costmap_pub->publishCostmap();
```

Comment:
Maybe this publisher should be part of the down sampler itself?

Reply:
Also then you wouldn't have to return a nonsensical costmap in the initialize function

Reply:
The publisher is always seeing the most updated version of the downsampled costmap. I give it a pointer upon creation, and then the object being pointed gets updated internally when calling  `_costmap_downsampler->downsample()`.

I guess I can make the publisher part of the downsampler, it would only require passing a bunch of variables to the initialize method of the downsampler (ros node, global frame and topic name)